prov/sockets/src/sock_progress.c

```diff
@@ -1169,6 +1169,7 @@ ssize_t sock_rx_claim_recv(struct sock_rx_ctx *rx_ctx, void *context,
 
 		dlist_remove(&rx_buffered->entry);
 		sock_rx_release_entry(rx_buffered);
+		rx_ctx->progress_start = &rx_ctx->rx_buffered_list;
```

Comment:
Can you please add more detail to the commit message as to why this change is needed and what it fixes?

Reply:
added some comments

Reply:
It fixed seg fault in a multi-threading test.